internal/source/git.go

```diff
@@ -38,9 +43,24 @@ func (r *Git) Unpack(ctx context.Context, bundle *rukpakv1alpha1.Bundle) (*Resul
 	// Set options for clone
 	progress := bytes.Buffer{}
 	cloneOpts := git.CloneOptions{
-		URL:      gitsource.Repository,
-		Progress: &progress,
-		Tags:     git.NoTags,
+		URL:             gitsource.Repository,
+		Progress:        &progress,
+		Tags:            git.NoTags,
+		InsecureSkipTLS: bundle.Spec.Source.Git.InsecureSkipVerify,
+	}
+
+	if bundle.Spec.Source.Git.Secret.Name != "" {
+		userName, password, err := r.getCredential(ctx, bundle)
+		if err != nil {
+			return nil, err
+		}
+		if password == "" {
+			return nil, errors.New("data.accessToken is missing")
+		}
+		if userName == "" {
+			userName = "user"
+		}
```

Comment:
nit: could we put this logic inside `getCredential` so it returns either the auth or an error?